File: android-app/app/src/main/java/com/kunlun/studentapp/data/model/UserModels.kt
package com.kunlun.studentapp.data.model

data class UserInfo(
    val id: Int,
    val username: String,
    val real_name: String,
    val role: String
)

data class MeSettings(
    val current_week: String,
    val current_month: String,
    val daily_limit: Int,
    val score_options: List<Double>,
    val weekly_max_score: Double
)

data class MeData(
    val user: UserInfo,
    val settings: MeSettings,
    val today_submit_count: Int,
    val recent_records: List<ScoreRecord>
)

File: ios-app/KunlunStudentApp/Models/UserModels.swift
import Foundation

struct UserInfo: Codable, Equatable {
    let id: Int
    let username: String
    let realName: String
    let role: String
}

struct MeSettings: Decodable {
    let currentWeek: String
    let currentMonth: String
    let dailyLimit: Int
    let scoreOptions: [Double]
    let weeklyMaxScore: Double
}

struct MeData: Decodable {
    let user: UserInfo
    let settings: MeSettings
    let todaySubmitCount: Int
    let recentRecords: [ScoreRecord]
}

